src/tests/Helpers/PlanHelperTest.php

```diff
@@ -0,0 +1,332 @@
+		
+		$this->template_loader_helper_mock = \Mockery::mock(TemplateLoaderHelper::class);
+		$this->template_loader_helper_mock->shouldReceive('get_template')->andReturn('');
+		
+		$this->plan_builder_helper = \Mockery::mock(PlanHelperBuilder::class)->makePartial();
+		$this->plan_builder_helper->shouldReceive('get_alma_settings')->andReturn($this->alma_settings_mock);
+		$this->plan_builder_helper->shouldReceive('get_gateway_helper')->andReturn($this->gateway_helper_mock);
+		$this->plan_builder_helper->shouldReceive('get_template_loader_helper')->andReturn($this->template_loader_helper_mock);
+		$this->plan_builder_helper->shouldReceive('get_price_factory')->andReturn($this->price_factory_mock);
+		
+		$this->alma_plan_helper = $this->plan_builder_helper->get_instance();
+
+	}
+
+	public function test_get_plans_by_keys_empty_plans() {
+		$this->assertEquals(array(), $this->alma_plan_helper->get_plans_by_keys());
+	}
+
+	public function test_get_plans_by_keys_empty_eligibilities() {
+		$this->assertEquals(array(), $this->alma_plan_helper->get_plans_by_keys(
+			 array(
+				 ConstantsHelper::PAY_NOW_FEE_PLAN,
+			 )
+		));
+	}
+
+	public function test_get_plans_by_keys_empty_eligibilities_plans() {
+		$this->assertEquals(
+			array(),
+			$this->alma_plan_helper->get_plans_by_keys(
+				array(),
+				array(
+					ConstantsHelper::PAY_NOW_FEE_PLAN => $this->eligibility_mock ,
+					ConstantsHelper::DEFAULT_FEE_PLAN => $this->eligibility_mock
+				)
+			));
+	}
+
+	public function test_get_plans_by_keys() {
+		$elegibility = \Mockery::mock(Eligibility::class);
+		$this->assertEquals(
+			array( ConstantsHelper::PAY_NOW_FEE_PLAN => $this->eligibility_mock ),
+			$this->alma_plan_helper->get_plans_by_keys(
+			array(
+					ConstantsHelper::PAY_NOW_FEE_PLAN,
+			),
+			array(
+				ConstantsHelper::PAY_NOW_FEE_PLAN => $this->eligibility_mock ,
+				ConstantsHelper::DEFAULT_FEE_PLAN => $this->eligibility_mock
+			)
+		));
+	}
+
+	public function test_order_plans_empty_array() {
+		$this->assertEquals(array(), $this->alma_plan_helper->order_plans(array()));
+	}
+	public function test_order_plans_empty_plans() {
+		$this->assertEquals(array(), $this->alma_plan_helper->order_plans(array(), ConstantsHelper::GATEWAY_ID));
+	}
+
+	public function test_order_plans_empty_gateway_id() {
+		$alma_settings_mock = \Mockery::mock(AlmaSettings::class)->makePartial();
+		$alma_settings_mock->settings['display_in_page'] = 'no';
+		$alma_settings_mock->shouldReceive('should_display_plan')->andReturn(false);
+
+		$plan_builder_helper = \Mockery::mock(PlanHelperBuilder::class)->makePartial();
+		$plan_builder_helper->shouldReceive('get_alma_settings')->andReturn($alma_settings_mock);
+		$plan_helper = $plan_builder_helper->get_instance();
+		$this->assertEquals(array(), $plan_helper->order_plans(array(ConstantsHelper::PAY_NOW_FEE_PLAN)));
+	}
+
+	public function test_order_plans_no_gateway_id_no_in_page_no_display_plan() {
```

Comment:
This test test the same case that the test before, the different param doesn't change the logic and the result of the test

Reply:
Fixed